include/beman/scope/scope.hpp

```diff
@@ -3,13 +3,48 @@
 #ifndef BEMAN_SCOPE_HPP
 #define BEMAN_SCOPE_HPP
 
+#include <concepts>
+#include <exception>
 #include <type_traits>
-#include <utility>
+#include <version>
+
+#if (!(defined(__cpp_concepts) || !(defined(__cpp_lib_concepts))))
+static_assert(false, "C++20 Concepts Required");
+#endif
```

Comment:
as coded this probably won't work -- you'd need to include version first and then do this check before doing the include of concepts.  With the include order above I think this will just fail with something like 'cant find the header concepts'.

Probably the easiest solution is to just remove the check and let the header failure be the error, but this is clearly nicer to the users if they make it this far.  Of course at this point the docs have already warned them and so has the cmake that they need c++20